### PR TITLE
Remove Emotion

### PR DESCRIPTION
**Pull request by alexpate**
Continuing along the theme of the UI library becoming as vanilla as possible, it makes sense to remove Emotion as a core library for handling styling.

CSS modules gives us 90% of what we're after (relative encapsulation, and file collocation), although we will still need a solution for enabling custom branding.

Users can set two of their own brand colors (primary/secondary), which are received from the API at runtime. CSS custom properties could be something to explore?

### CODE REVIEW

**Comment by sirLisko** [comment]
All the references to `emotion` and `react-emotion` have been removed.

To support the `primary` *brand colour* (the only one used here) two CSS variables are available: 

```
--ui-user-primary
--ui-user-primary--lighter
```

In order to use them, they need to be defined, otherwise, they will fall back to the default primary colour.
```
<div
  style={{
    '--ui-user-primary': '#0d7ef2',
    '--ui-user-primary--lighter': '#cad9e8',
  }}
>
  <Button variant="primary" size="large">
    A themed button
  </Button>
</div>
```

To support style props (i.e. `alignItem="center"`) [`inline-style-prefixer`](https://github.com/rofrischmann/inline-style-prefixer) has been introduced (more details https://github.com/kalohq/ui/pull/270/commits/e3347b59801d6e029a23570733ed8effd0e3e8b0).

For the future, ideally, we would like to get rid of the support to these style props, in order to remove the `prefixer` dependency and leave the app using the `ui` dealing with it.

File: site/gatsby-node.js [hunk 1]
```diff
@@ -129,7 +129,10 @@ exports.createPages = ({boundActionCreators, graphql}) => {
   const componentPageTemplate = path.resolve(`src/templates/component.js`);
   return graphql(`
     {
-      allMarkdownRemark(sort: {fields: [fields___componentName], order: ASC}, filter: {fields: {ignorePage: {eq: false}}}) {
+      allMarkdownRemark(
+        sort: {fields: [fields___componentName], order: ASC}
+        filter: {fields: {ignorePage: {eq: false}}}
+      ) {
```
Comment (olleolleolle) [inline site/gatsby-node.js]: Improved formatting!

File: src/components/avatar-group/__tests__/avatar-group.test.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env jest */
 import * as React from 'react';
 
-import {shallow, mount} from 'utils/test/enzyme';
+import {shallow, mount} from 'enzyme';
```
Comment (olleolleolle) [inline src/components/avatar-group/__tests__/avatar-group.test.js]: More vanilla! Improvement!

File: src/components/fieldset/fieldset.react.js [hunk 3]
```diff
@@ -15,42 +18,33 @@ export function FieldsetHeader(props: {
   const {legend, description, meta} = props;
 
   return (
-    <Box marginTop={20}>
-      {legend ? (
+    <UIBase marginTop={20}>
+      {legend && (
```
Comment (olleolleolle) [inline src/components/fieldset/fieldset.react.js]: Conditionals: an improvement over `what ? <Then> : null`!

File: src/components/layout/README.md [hunk 4]
```diff
@@ -3,35 +3,34 @@
...
-marginLeft="extra-small" // margin-left: 2px
-marginLeft="small" // margin-left: 4px
-marginLeft="medium" // margin-left: 8px
-marginLeft="large" // margin-left: 16px
-marginLeft="extra-large" // margin-left: 24px
+marginLeft = 'extra-small'; // margin-left: 2px
+marginLeft = 'small'; // margin-left: 4px
+marginLeft = 'medium'; // margin-left: 8px
+marginLeft = 'large'; // margin-left: 16px
+marginLeft = 'extra-large'; // margin-left: 24px
```
Comment (olleolleolle) [inline src/components/layout/README.md]: Improvement in formatting!

File: src/components/table/table-cell.react.js [hunk 5]
```diff
@@ -101,16 +31,27 @@ type TTableCellProps = {
 };
 
 export function TableCell(props: TTableCellProps) {
-  const {active, children, header, order, ...otherProps} = props;
-  const style = pickStyles(otherProps);
+  const {active, children, header, order, className, ...otherProps} = props;
   const icon = `keyboard_arrow_${order === 'desc' ? 'down' : 'up'}`;
-  const Component = header ? StyledTableHeaderCell : StyledTableCell;
+  // const Component = header ? StyledTableHeaderCell : StyledTableCell;
```
Comment (olleolleolle) [inline src/components/table/table-cell.react.js]: Q: Is this code commented out to stay?

**Comment by oliverlloyd** [comment]
I've read thorugh the changes - breifly, there are a lot - and not seen any glaring problems. I've also run npm installs and a build locally and the Gatsby site runs fine for me.

👍

File: src/components/table/table-cell.react.js (hunk 5 above)
Comment (sirLisko) [reply src/components/table/table-cell.react.js]: Has been fixed in the flow removal branch 👍

**Comment by sirLisko** [comment]
A problem has been found when the prefixer is returning an array of values (i.e. `display: 'inline-flex'`).
To solve this issue we are falling back to the default prop value (for more details https://github.com/kalohq/ui/commit/70a59c55c6088c2e544210438e4afc53358247a4)

```
     style={
          Object {
            "WebkitBoxAlign": "center",
            "alignItems": "center",
    -       "display": Array [
    -         "-webkit-inline-box",
    -         "-moz-inline-box",
    -         "-ms-inline-flexbox",
    -         "-webkit-inline-flex",
    -         "inline-flex",
    -       ],
    +       "display": "inline-flex",
          }
```

**Pull request merged**